### PR TITLE
Upgrade `electron` to `v22.0.1`

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/ferdium/ferdium-recipes/issues)
2. If you are requesting support for a **new service** in Ferdium, please log it [here](https://github.com/ferdium/ferdium-recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/ferdium/ferdium-app#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdium, but to get new recipes between Ferdium releases, this documentation is quite useful.)
4. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/ferdium/ferdium-app/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change

Upgrading to the latest beta version of electron to see if we need to make any code fixes. Looks like we have to fix the deprecated method that was removed.

#### Motivation and Context

Fix any deprecated methods, and other upgrades if needed to move to the next electron version when it gets released.

#### Screenshots

Screenshot 2022-12-08 at 4 29 32 AM

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

**Comment by vraravam** [comment]
@balajiv113 / @muhamedsalih-tw - need some help here. All other services other than google's are working correctly. Please review and help fix.

**Comment by balajiv113** [comment]
Fixed the issue with Google
@vraravam can you give a try now

**Comment by balajiv113** [comment]
We can even fork react-electron-web-view and do this changes in our fork as well. If this is a preferred option

**Comment by vraravam** [comment]
> We can even fork react-electron-web-view and do this changes in our fork as well. If this is a preferred option

We should keep [this discussion](https://github.com/ferdium/ferdium-app/discussions/819) in view and make whichever decision that can be beneficial to the project in the longer run.

**Comment by vraravam** [comment]
> Fixed the issue with Google @vraravam can you give a try now

wow! Works great now!

**Comment by vraravam** [comment]
oops - not sure how, but i have 2 gmail services configured: one for TW, and one for personal. With this new version, both services are somehow using the same session. The sandboxing is broken!

**Comment by vraravam** [comment]
oops - not sure how, but i have 2 gmail services configured: one for TW, and one for personal. With this new version, both services are somehow using the same session. The sandboxing is broken!
Quitting and opening the 6.2.2 stable release on the same Ferdium profile folder resurrects the sandboxed sessions correctly.

**Comment by balajiv113** [comment]
> > We can even fork react-electron-web-view and do this changes in our fork as well. If this is a preferred option
> 
> We should keep [this discussion](https://github.com/ferdium/ferdium-app/discussions/819) in view and make whichever decision that can be beneficial to the project in the longer run.

I believe here its more about having those Dompurify code changes of that dependency. Whether those changes should be done in fork / copied to our source 

To me fork looks good as converting that class to tsx was super hard it took more than 2hrs and yet i failed to do it 😁

**Comment by vraravam** [comment]
> To me fork looks good

in that case, lets do a fork and raise a PR for the upstream to take in our changes.

**Comment by balajiv113** [comment]
Found the core cause. if we enabled nodeintegration for webview it is crashing. So disabled it. 
Not sure if we need. if needed we need to wait for electron to fix it

File: src/index.ts [hunk 1]
```diff
@@ -397,10 +396,9 @@ const createWindow = () => {
   // @ts-expect-error Property 'isMaximized' does not exist on type 'App'.
   app.isMaximized = mainWindow.isMaximized();
 
-  mainWindow.webContents.on('new-window', (e, url) => {
-    e.preventDefault();
-    openExternalUrl(url);
-  });
+  mainWindow.webContents.setWindowOpenHandler(() => ({
```
Comment (balajiv113) [inline src/index.ts]: I think the flow here looks wrong.

We should trigger openExternalUrl(url) and send {action: "deny"} right ?

Comment (vraravam) [reply src/index.ts]: yes - could you please make the change?
Also, i dont see the unread message count in the sidebar with this branch.

Comment (balajiv113) [reply src/index.ts]: Sure will check

Comment (vraravam) [reply src/index.ts]: yes - also context menu is missing and clicking on links doesn't open the default external browser

Comment (vraravam) [reply src/index.ts]: @balajiv113 - now that we have backported some of the fixes (for deprecation), I see that only the `nodeIntegration` has been removed after the major version upgrade of electron in this PR. Based on this, can we conclude that we need to file a bug report against electron to get this sorted before proceeding with any more upgrades of electron?

Comment (balajiv113) [reply src/index.ts]: Yes yes exactly. I created one issue already in electron with a simple reproduction repo as well

https://github.com/electron/electron/issues/36608

**Comment by vraravam** [comment]
Blocked waiting for https://github.com/electron/electron/issues/36608 to be released in a stable version in v22.x series.

File: src/index.ts (hunk 1 above)
Comment (vraravam) [reply src/index.ts]: This has been just released last night in electron v22.0.1. Closing

**Comment by vraravam** [comment]
Though my last commit re-introduces the `nodeintegration` configuration, @balajiv113 - do you think we can completely remove that? Does anyone else have any idea why this was required?

**Comment by vraravam** [comment]
> Though my last commit re-introduces the `nodeintegration` configuration, @balajiv113 - do you think we can completely remove that? Does anyone else have any idea why this was required?

np - figured it out! Looks like this is required for the unread badges to work

File: package.json [hunk 2]
```diff
@@ -151,7 +151,8 @@
     "chalk": "5.2.0",
     "concurrently": "7.6.0",
     "cross-env": "7.0.3",
-    "electron": "21.3.3",
+    "dotenv": "16.0.3",
```
Comment (vraravam) [inline package.json]: @balajiv113 - any reason why we introduced this dependency?

Comment (balajiv113) [reply package.json]: I believe i added it during the migration. Looks like not needed only.

**Review by balajiv113 — APPROVED** [review]
LGTM

**Pull request merged**